src/main/java/com/spotify/docker/client/messages/swarm/Service.java

```diff
@@ -71,10 +71,37 @@
     @Nullable
     public abstract String serviceName();
 
+    /**
+     * Filter by label.
+     */
+    @Nullable
+    public abstract Label label();
+    
     public static Builder builder() {
       return new AutoValue_Service_Criteria.Builder();
     }
 
+    @AutoValue
+    public abstract static class Label {
+      
+      public abstract String name();
+      
+      public abstract String value();
+      
+      public static Builder builder() {
+        return new AutoValue_Service_Criteria_Label.Builder();
+      }
+      
+      @AutoValue.Builder
+      public abstract static class Builder {
```

Comment:
Does Label really need its own Builder? Seems like it would be easier to just have a static method `Service.Builder.label(String, String)` that makes a new Label and adds it to the Service Builder. Maybe also a second one that takes only one String if you just want a Label with name and no value.